prov/verbs/src/verbs_mr.c

```diff
@@ -65,17 +65,11 @@ fi_ibv_mr ## type ## reg(struct fid *fid, const void *buf, size_t len,			\
 					 requested_key, flags, mr, context);		\
 }											\
 											\
-static struct fi_ops_mr fi_ibv_domain_mr ##type## ops = {				\
+struct fi_ops_mr fi_ibv_mr ##type## ops = {						\
 	.size = sizeof(struct fi_ops_mr),						\
 	.reg = fi_ibv_mr ## type ## reg,						\
 	.regv = fi_ibv_mr ## type ## regv,						\
 	.regattr = fi_ibv_mr ## type ## regattr,					\
```

Comment:
Because it hides the function names?

Reply:
It just obscures the code.  We don't need to generate different functions for mr_reg and mr_regv based on whether or not there is a cache.  They do the same thing and can just call fi_mr_regv() and fi_mr_regattr(), respectively.  Only the regattr call differs.  We just need something like:
```
fi_ibv_mr_reg(...)
{
    ...
    return fi_mr_regv(...);
}

fi_ibv_mr_regv(...)
{
    ....
    return fi_mr_regattr(...)
}

fi_ibv_mr_regattr(...)
{
    ....
}

fi_ibv_mr_cache_regattr(...)
{
    ...
}
``

Reply:
Actually, since verbs only supports a single buffer, it would be better to make the single buffer calls the main entry points instead of regattr.
```
fi_ibv_mr_reg(...)
{
    return ibv_reg_mr(...)
}
fi_ibv_mr_cache_reg(...)
{
    .. handle cached registrations...
}
fi_ibv_regv(...)
{
    return fi_mr_reg(...)
}
etc.
```